drafts/draft-ietf-ippm-ioam-ipv6-options.xml

```diff
@@ -0,0 +1,499 @@
+      independent of whether the Hop-by-Hop options or Destination options are
+      used to encode the IOAM data. This ensures that IOAM data does not
+      unintentionally get forwarded outside the IOAM domain.</t>
+
+      <t>An IPv6 packet carrying IOAM data in an Extension header can have
+      other extension headers, compliant with <xref
+      target="RFC8200"></xref>.</t>
+
+      <t>IPv6 Hop-by-Hop and Destination Option format for carrying in-situ
+      OAM data fields:<figure>
+          <artwork><![CDATA[
+
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|  Option Type  |  Opt Data Len |   Reserved    |   IOAM Type   |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+|                                                               |  |
+.                                                               .  I
+.                     Option Data                               .  O
+.                                                               .  A
+.                                                               .  M
+.                                                               .  .
+.                                                               .  O
+.                                                               .  P
+.                                                               .  T
+.                                                               .  I
+.                                                               .  O
+.                                                               .  N
+.                                                               .  |
+|                                                               |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+
+]]></artwork>
+        </figure></t>
+
+      <t><list style="hanging">
+          <t hangText="Option Type:">8-bit identifier of the type of
+          option.</t>
+
+          <t hangText="Opt Data Len:">8-bit unsigned integer. Length of the
+          Reserved and Option Data field of this option, in octets.</t>
+
+          <t hangText="Reserved:">8-bit field MUST be set to zero upon
+          transmission and ignored upon reception.</t>
+
+          <t hangText="IOAM Type:">8-bit field as defined in section 7.2 in
+          <xref target="I-D.ietf-ippm-ioam-data"></xref>.</t>
+
+          <t hangText="Option Data:">Variable-length field.
+          Option-Type-specific data.</t>
+        </list></t>
+
+      <t>In-situ OAM Options are inserted as Option data as follows:<list
+          style="numbers">
+          <t>Pre-allocated Tracing Option: The in-situ OAM Preallocated
+          Tracing option defined in <xref
+          target="I-D.ietf-ippm-ioam-data"></xref> is represented as a IPv6
+          option in hop by hop extension header: <list style="hanging">
+              <t hangText="Option Type:">001xxxxx 8-bit identifier of the IOAM
```

Comment:
Why start with "00" if the default behavior is to drop packets which contain extension headers carrying IOAM data-fields?
"00" means when the option type is unrecognized, "skip over this option and continue processing the header".
The other possibilities, any of which would be consistent with the default behavior described above, are:

- "01" (discard the packet),
- "10" (discard and send ICMP parameter problem, code 2, back to the packet's source address), or
- "11" (discard and send ICMP only if the packet's destination address was not a multicast address).

Reply:
Sorry, draft-ietf-ippm-ioam-ipv6-options.xml changes showed up in the pull request as my local master was not in sync with the repo. Now merged to latest. Only intended to address data draft changes in this pull request.